### PR TITLE
Main module refactoring (part1)

### PR DESCRIPTION
**Pull request by pashutk**
Added jsdocs, separate module for Layer class, some refactoring

### CODE REVIEW

File: src/modV.js [hunk 1]
```diff
@@ -1,53 +1,59 @@
...
 require('./fragments/array-contains');
 require('script-loader!../libraries/beatdetektor.js');
 
+
 /**
  * modV
  * @extends {EventEmitter2}
  */
-class modV extends EventEmitter2 {
+class ModV extends EventEmitter2 {
```
Comment (2xAA) [inline src/modV.js:15]: Although is it convention to use an uppercase first symbol on classes, modV should be kept lowercase.
It is its name and every modV Moduletype depends upon this

Comment (2xAA) [reply src/modV.js:15]: (this is the only modV class which will use a lowercase first symbol, modV was created as a singleton)

Comment (pashutk) [reply src/modV.js:15]: But it's just a class name, and in `app.js` you can see line: `const modV = require('./modV');`.
What's wrong?

Comment (pashutk) [reply src/modV.js:15]: Hmm, I think now i see what's the problem. I'll try to fix that, or rename the class.

Comment (pashutk) [reply src/modV.js:15]: Resolved

Comment (2xAA) [reply src/modV.js:15]: (for history) I think it's fine, naming the class with an uppercase symbol is perfectly fine, it does differentiate between the class and instance which can be useful.
Let's make sure in the documentation we name `ModV` as `modV` however as it is presented to the user as `modV`

**Pull request merged**